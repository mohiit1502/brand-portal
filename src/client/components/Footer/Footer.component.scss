@import "./../../styles/override/variables.scss";

.c-Footer {
  //padding: 1.75rem 0;
  //padding: 0 6rem;
  background: #286697;
  color: white;
  overflow: hidden;

  .c-Footer__logoSocialContainer {
    display: inline-block;
    position: relative;
    //margin-bottom: 0.6rem;

    .c-Footer__wmWhiteLogo {
      //width: 7.625rem;
      height: 30px;
    }

    .c-Footer__logoText {
      bottom: -0.75rem;
      font-size: 0.8rem;
      line-height: 1.125rem;
      position: absolute;
      right: 2.1rem;
    }
  }

  .c-Footer__copyrightText {
    padding: 10px 6rem;
    color: #eee;
    //padding-left: 6rem;
    .c-Footer__privacyNav {
        color: #eee;
      margin: 0 10px;
      //border-left: 2px solid white;
      //padding-left: 10px;
    }
    .c-Footer__privacyNav:before{
      content: "|";
      padding-right: 10px;
    }

  }

  .login-footer-nav {
    padding: 1rem 6rem;
    background: #235882;

    .navbar-collapsible-footer {
      .nav-item {
        .nav-link {
          font-size: 1rem;
          color: $white;
          cursor: pointer;

          &.login-btn {
            // width : 90px;
            border: 1px solid $white;
            padding: 6px 25px;

            &:hover {
              background: $white;
              color: $blue;
              border-color: $blue;
            }
          }

          &:hover {
            color: $white;
            font-weight: bolder;
          }
        }
      }
    }
  }
}
