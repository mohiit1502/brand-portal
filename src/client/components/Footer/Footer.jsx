import React from "react";
import * as images from "./../../images";
import CONSTANTS from "./../../constants/constants";
import "./Footer.component.scss";
import WMBPDarkLogo from "./../../images/WMWhite-horizontal.svg";
import mixpanel from "../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../constants/mixpanelConstants";

const Footer = props => {

  const scrollAction = () => {
    $("#LoginFAQ").addClass("scrolled");
    setTimeout(() => $("#LoginFAQ").removeClass("scrolled"), 1000);
  };

  return (
    <div className="c-Footer">
      <div className="row">
        <div className="col">
          <nav className="navbar navbar-expand-lg login-footer-nav">
            <div className="c-Footer__logoSocialContainer">
              <img className="c-Footer__wmWhiteLogo" src={WMBPDarkLogo} />
              {/* <span className="c-Footer__logoText">Walmart Brand Portal</span> */}
            </div>
            <div className="collapse navbar-collapse navbar-collapsible-footer" id="collapsible-footer">
              <ul className="navbar-nav ml-auto">
                {/* <li className="nav-item mx-4">
                  <a className="nav-link" href="#">Home</a>
                </li> */}
                <li className="nav-item mx-1 d-flex align-items-center">
                  <a className="nav-link btn login-btn" href={CONSTANTS.URL.LOGIN_REDIRECT} onClick={() => {mixpanel.trackEvent(MIXPANEL_CONSTANTS.HOME_PAGE_EVENTS.CLICK_ON_LOGIN);}}>Login</a>
                  {/*<img className="nav-link" src={youtube}/>*/}
                </li>
                <li className="nav-item ml-1">
                  <a className="nav-link" href={CONSTANTS.URL.REGISTER_REDIRECT} onClick={() => {mixpanel.trackEvent(MIXPANEL_CONSTANTS.HOME_PAGE_EVENTS.CLICK_ON_REGISTER);}}>Register</a>
                  {/*<img className="nav-link" src={walmart}/>*/}
                </li>
              </ul>
            </div>
          </nav>
        </div>
      </div>
      <div className="row">
        <div className="col">
          <div className="c-Footer__copyrightContainer">
            <span className="c-Footer__copyrightText float-left">
              {CONSTANTS.LOGIN.COPYRIGHTTEXT}
              <a target="_blank" href={CONSTANTS.LOGIN.PRIVACYURL} className="c-Footer__privacyNav float-right">{CONSTANTS.LOGIN.PRIVACYTEXT}</a>
            </span>
            {/*<a target="_blank" href={CONSTANTS.LOGIN.PRIVACYURL} className="c-Footer__privacyNav float-right">{CONSTANTS.LOGIN.PRIVACYTEXT}</a>*/}
          </div>
        </div>
      </div>
    </div>
  );
};

Footer.propTypes = {

};

export default Footer;
