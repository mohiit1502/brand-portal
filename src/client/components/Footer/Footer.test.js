/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import Footer from "./Footer";

describe("Footer", () => {
   it("renders without error", () => {

   });
});
