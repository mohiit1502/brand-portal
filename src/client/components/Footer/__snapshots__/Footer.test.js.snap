// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockNextContext
    pathname="/dashboard"
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/dashboard",
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/dashboard",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/dashboard",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Footer
          isWebform={false}
        >
          <div
            className="c-Footer"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <nav
                  className="navbar navbar-expand-lg login-footer-nav"
                >
                  <div
                    className="c-Footer__logoSocialContainer"
                  >
                    <img
                      className="c-Footer__wmWhiteLogo"
                      src="WMWhite-horizontal.svg"
                    />
                  </div>
                  <div
                    className="collapse navbar-collapse navbar-collapsible-footer"
                    id="collapsible-footer"
                  >
                    <ul
                      className="navbar-nav ml-auto"
                    >
                      <li
                        className="nav-item mx-1 d-flex align-items-center"
                      >
                        <Link
                          className="nav-link btn login-btn py-1"
                          onClick={[Function]}
                          to="/login"
                        >
                          <LinkAnchor
                            className="nav-link btn login-btn py-1"
                            href="/login"
                            navigate={[Function]}
                            onClick={[Function]}
                          >
                            <a
                              className="nav-link btn login-btn py-1"
                              href="/login"
                              onClick={[Function]}
                            >
                              Login
                            </a>
                          </LinkAnchor>
                        </Link>
                      </li>
                      <li
                        className="nav-item ml-1"
                      >
                        <Link
                          className="nav-link"
                          onClick={[Function]}
                          to="/register"
                        >
                          <LinkAnchor
                            className="nav-link"
                            href="/register"
                            navigate={[Function]}
                            onClick={[Function]}
                          >
                            <a
                              className="nav-link"
                              href="/register"
                              onClick={[Function]}
                            >
                              Register
                            </a>
                          </LinkAnchor>
                        </Link>
                      </li>
                    </ul>
                  </div>
                </nav>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="c-Footer__copyrightContainer"
                >
                  <span
                    className="c-Footer__copyrightText float-left"
                  >
                    Â© 2021 Walmart. All rights reserved.
                    <a
                      className="c-Footer__privacyNav float-right"
                      href="https://corporate.walmart.com/privacy-security"
                      target="_blank"
                    >
                      Privacy Policy
                    </a>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </Footer>
         
      </Router>
    </MemoryRouter>
  </MockNextContext>
</Provider>
`;
